.travis.yml

```diff
@@ -4,43 +4,36 @@ before_cache:
     - rm -fr ./target/debug/{deps,.fingerprint}/{*ra_*,*test*,*tools*,*gen_lsp*,*thread_worker*}
     - rm -f  ./target/.rustc_info.json
 
-build: &rust_build
-    language: rust
-    rust: stable
-    script:
-        - rustup component add rustfmt
-        - rustup component add rust-src
-        - cargo test --no-run  # let's measure compile time separately
-        - cargo test
-        - cargo doc --all --no-deps
-    env:
-        - RUSTFLAGS="-D warnings", CARGO_INCREMENTAL=0
-
 matrix:
     include:
         - os: linux
-          before_script:
+          language: rust
+          rust: stable
+          script:
+              - rustup component add rustfmt
+              - rustup component add rust-src
+              - cargo test --no-run  # let's measure compile time separately
+              - cargo test
+          env:
+              - RUSTFLAGS="-D warnings", CARGO_INCREMENTAL=0
+
+        - os: linux
+          if: branch = master AND type = push
+          before_stript:
```

Comment:
```suggestion
          before_script:
```

Reply:
wow, that wouldn't have been caught for ages!